Update build/auth documentation to document kubernetes.io/* secrets (#1098)
knative/docs: docs/build/auth.md

be generated containing the credentials configured in the `Secret`, and these
credentials are then used to authenticate with the Docker registry.

> ## Kubernetes's docker registry's secret

Reviewer: ```suggestion
## Kubernetes's Docker registry's secret
```

> Kubernetes defines two type of secret for docker registries :

Reviewer: ```suggestion
Kubernetes defines two types of secrets for Docker registries:
```

`kubernetes.io/dockercfg` and
`kubernetes.io/dockerconfigjson`. Knative supports those secrets in
addition to the one described above.

> 1. Define a `Secret` from a docker client configuration file, as

Reviewer: ```suggestion
1. Define a `Secret` from a Docker client configuration file, as
```
